packages/core/src/components/event-lake.aggregate-store.suite.ts

```diff
@@ -0,0 +1,336 @@
+    };
+
+    await expectMS(5000, async () => {
+      for (const account of accounts) {
+        await accountStore.save(account);
+      }
+    });
+
+    await expectMS(5000, async () => {
+      for (const account of accounts) {
+        await accountStore.load(account.id);
+      }
+    });
+  });
+
+  it("supports saveAll with transactions", async () => {
+    const accountStore = config.getAggregateStore(Account, registry);
+
+    const accounts = [...Array(10).keys()].map(() => Account.open());
+    await accountStore.saveAll(accounts);
+
+    const loadedAccounts = await Promise.all(
+      accounts.map((account) => accountStore.load(account.id)),
+    );
+
+    expect(loadedAccounts).toEqual(accounts);
+  });
+
+  class MockEventBus implements IEventBus {
+    off(): void {
+      throw new Error("Method not implemented.");
+    }
+    on() {
+      throw new Error("Method not implemented.");
+    }
+
+    events: IEvent[] = [];
+
+    publish(event: IEvent) {
+      this.events.push(event);
+      return Promise.resolve();
+    }
+  }
+  it("should support saveAll with transactions", async () => {
+    const eventBus = new MockEventBus();
+    const accountStore = config.getAggregateStore(Account, registry, eventBus);
+
+    const bank = Bank.new();
+    await bankStore.save(bank);
+
+    let willFailForAttempt = 3;
+    const saved = await config.transaction.perform(async (trx) => {
+      willFailForAttempt--;
+      const trxBank = await bankStore.load(bank.id, trx);
+
+      const accounts = [...Array(30).keys()].map((i) =>
+        Account.openIndexed(
+          bank.id,
+          trxBank!.increment(willFailForAttempt === 0),
+        ),
+      );
+
+      await accountStore.saveAll(accounts, trx);
+      await bankStore.save(trxBank!, trx);
+      return accounts;
+    });
+
+    const freshBank = await bankStore.load(bank.id);
+
+    expect(freshBank!.accounts).toBe(30);
+
+    const freshAccounts = (
+      await Promise.all(saved.map((a) => accountStore.load(a.id)))
+    ).filter((a) => !!a) as Account[];
+
+    expect(freshAccounts.length).toBe(30);
+
+    const indices = freshAccounts
+      .map((doc) => Number(doc.index))
+      .sort((a, b) => a - b);
+
+    expect(indices).toEqual([
+      1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21,
+      22, 23, 24, 25, 26, 27, 28, 29, 30,
+    ]);
+
+    expect(eventBus.events.length).toBe(30);
+  });
+
+  it("should support crosslocking with transactions", async () => {
+    const eventBus = new MockEventBus();
+    const accountStore = config.getAggregateStore(Account, registry, eventBus);
+
+    const bank = Bank.new();
+    await bankStore.save(bank);
+
+    const accountIds = await Promise.all([
+      config.transaction.perform(async (trx) => {
+        const bk = await bankStore.load(bank.id, trx);
+
+        await new Promise((resolve) => setTimeout(resolve, 100));
+
+        const account = Account.openIndexed(bank.id, bk!.increment());
```

Comment:
can you put different account values in the two transactions and check wich index correspond to which account. (to see which transaction is resolved first)

Reply:
is it also possible to see how many time each transaction has been tried ?